lib/utils.js

```diff
@@ -7,6 +7,19 @@ function unique(array) {
   return [...new Set(array)];
 }
 
+/**
+ * Convert anything into an error.
+ *
+ * If it's already an error, do nothing.
+ * Otherwise, wrap it in a generic error.
+ *
+ * @param {unknown} error
+ */
+function intoError(error) {
+  return error instanceof Error ? error : new Error(String(error));
```

Comment:
I'm curious, did you notice places where we (potentially) wrapped an exception into multiple `Error`s? (nested I mean)

Reply:
> I'm curious, did you notice places where we (potentially) wrapped an exception into multiple `Error`s? (nested I mean)

Nope.

> Related: did you notice places where we throw non-Errors? 🤔 (Note, I haven't read the rest of the commits yet)

Yeah, we throw strings in a few places.